modules/apps/batch-engine/batch-engine-service/src/main/java/com/liferay/batch/engine/internal/instance/lifecycle/MultiCompanyBatchEngineUnitPortalInstanceLifecycleListener.java

```diff
@@ -27,11 +28,13 @@ public class MultiCompanyBatchEngineUnitPortalInstanceLifecycleListener
 
 	@Override
 	public void portalInstanceRegistered(Company company) throws Exception {
-		CompletableFuture<Void> completableFuture =
-			_multiCompanyBatchEngineUnitProcessor.processBatchEngineUnits(
-				company);
+		if (PortalInstancePool.getWebId(company.getCompanyId()) != null) {
```

Comment:
Mmm, this is a bit weird... I think this code should be executed anyway, as when a company is registered, everything related should be created too. By now, that would be protected by the FF

Reply:
Removed from now (let's hope it does not cause issues on CI :crossed_fingers:).